gen.go

```diff
@@ -938,29 +946,38 @@ func emitCborUnmarshalStructField(w io.Writer, f Field) error {
 
 func emitCborUnmarshalInt64Field(w io.Writer, f Field) error {
 	return doTemplate(w, f, `{
-	maj, extra, err := {{ ReadHeader "cr" }}
-	var extraI int64
+	{{ if .Pointer }}
+	b, err := cr.ReadByte()
 	if err != nil {
 		return err
 	}
-	switch maj {
-	case cbg.MajUnsignedInt:
-		extraI = int64(extra)
-		if extraI < 0 {
-			return fmt.Errorf("int64 positive overflow")
-	   }
-	case cbg.MajNegativeInt:
-		extraI = int64(extra)
-		if extraI < 0 {
-			return fmt.Errorf("int64 negative overflow")
-		}
-		extraI = -1 - extraI
-	default:
-		return fmt.Errorf("wrong type for int64 field: %d", maj)
-	}
+	if b != cbg.CborNull[0] {
+		if err := cr.UnreadByte(); err != nil {
+			return err
+		}
+{{ end }}maj, extra, err := {{ ReadHeader "cr" }}
+		if err != nil {
+			return err
+		}
+		var extraI int64
+		switch maj {
+		case cbg.MajUnsignedInt:
+			extraI = int64(extra)
+			if extraI < 0 {
+				return fmt.Errorf("int64 positive overflow")
+			}
+		case cbg.MajNegativeInt:
+			extraI = int64(extra)
+			if extraI < 0 {
+				return fmt.Errorf("int64 negative overflow")
+			}
+			extraI = -1 - extraI
+		default:
+			return fmt.Errorf("wrong type for int64 field: %d", maj)
+		}
 
-	{{ .Name }} = {{ .TypeName }}(extraI)
-}
+		{{ .Name }} = {{ if .Pointer }}(*{{ .TypeName }})(&extraI){{ else }}{{ .TypeName }}(extraI){{ end }}
```

Comment:
style nit: I'd break this over multiple lines (can duplicate the assignment, it's not much)